src/main/java/se/sundsvall/datawarehousereader/api/model/customer/CustomerDetails.java

```diff
@@ -0,0 +1,285 @@
+package se.sundsvall.datawarehousereader.api.model.customer;
+
+import static io.swagger.v3.oas.annotations.media.Schema.AccessMode.READ_ONLY;
+
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import se.sundsvall.datawarehousereader.api.model.CustomerType;
+
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+@Schema(description = "Customer details model")
+public class CustomerDetails {
+    @Schema(description = "Customer organization number, for internal use only", example = "5534567890", accessMode = READ_ONLY, hidden = true)
+    private String customerOrgNumber;
+    @Schema(description = "PartyId (e.g. a personId or an organizationId)", example = "81471222-5798-11e9-ae24-57fa13b361e1", accessMode = READ_ONLY)
+    private String partyId;
+    @Schema(description = "Customer number", example = "39195", accessMode = READ_ONLY)
+    private String customerNumber;
+    @Schema(description = "Customer name", example = "Sven Svensson", accessMode = READ_ONLY)
+    private String customerName;
+    @Schema(description = "Street", example = "Storgatan 44", accessMode = READ_ONLY)
+    private String street;
+    @Schema(description = "Postal code", example = "85230", accessMode = READ_ONLY)
+    private String postalCode;
+    @Schema(description = "City", example = "Sundsvall", accessMode = READ_ONLY)
+    private String city;
+    @Schema(description = "Care of address", example = "Agatha Malm", accessMode = READ_ONLY)
+    private String careOf;
+    @ArraySchema(schema = @Schema(description = "List of phoneNumbers", example = "076-1234567", accessMode = READ_ONLY))
+    private List<String> phoneNumbers;
+    @ArraySchema(schema = @Schema(description = "List of emailAddresses", example = "test@test.se", accessMode = READ_ONLY))
+    private List<String> emails;
+
+    @Schema(description = "Customer category ID", example = "1", accessMode = READ_ONLY)
+    private int customerCategoryID;
+
+    @Schema(description = "Customer category description", example = "Privat", accessMode = READ_ONLY)
+    private String customerCategoryDescription;
+
+    @Schema(description = "Indicates if customer details have changed since the search date", example = "PRIVATE", accessMode = READ_ONLY)
+    private boolean customerChangedFlg;
+
+    @Schema(description = "Indicates if placement and/or equipment details have changed since the search date", example = "true", accessMode = READ_ONLY)
+    private boolean installedChangedFlg;
+
+    @JsonIgnore
+    private CustomerType customerType;
```

Comment:
Is it necessary to store this attribute in the api class? It's used in the service layer, but the service layer also has access to the entity class (where this information can also be retreived from), so it might as well use that class instead, right?

Reply:
see above

Reply:
CustomerType is not part of the CustomerEngagement class, the logic to determin the type of customer could as easily be done where it is used instead of using the API class as a temporary (unneccessary) data holder. That is:

- Remove CustomerType and its getters and setters from API-class
- Remove mapping of customerType in CustomerDetailsMapper
- Make method extractCustomerType public (or move it to CustomerService class
- Change row 84-88 in CustomerService from:
```
		customerDetails.forEach(details -> {
			details.setPartyId(fetchPartyId(details.getCustomerType(), details.getCustomerOrgNumber()));
			details.setCustomerOrgNumber(null);

		});
```
to
```
		customerDetails.forEach(details ->
			details
				.withPartyId(fetchPartyId(extractCustomerType(details.getCustomerOrgNumber()), details.getCustomerOrgNumber()))
				.withCustomerOrgNumber(null));
```